test/acceptance/framework/consul_cluster.go

```diff
@@ -144,29 +143,25 @@ func (h *HelmCluster) SetupConsulClient(t *testing.T, secure bool) *api.Client {
 	remotePort := 8500 // use non-secure by default
 
 	if secure {
-		// overwrite remote port to HTTPS
+		// Overwrite remote port to HTTPS.
 		remotePort = 8501
 
-		// get the CA
-		caSecret, err := h.kubernetesClient.CoreV1().Secrets(namespace).Get(h.releaseName+"-consul-ca-cert", metav1.GetOptions{})
-		require.NoError(t, err)
-		caFile, err := ioutil.TempFile("", "")
-		require.NoError(t, err)
-		helpers.Cleanup(t, h.noCleanupOnFailure, func() {
-			require.NoError(t, os.Remove(caFile.Name()))
-		})
-
-		if caContents, ok := caSecret.Data["tls.crt"]; ok {
-			_, err := caFile.Write(caContents)
+		// Get the ACL token. First, attempt to read it from the bootstrap token.
+		// If the bootstrap token doesn't exist, try to read the replication token from the federation secret.
```

Comment:
Can we elaborate this comment a little bit as to why the token might not exists and potentially why it is ok for it to not be present.

Reply:
Yes, will do. good call!